server/app/services/DefectService.ts

```diff
@@ -40,6 +61,20 @@ export class DefectService {
     return defect;
   }
 
+  public async updateDefect(body: IDefect): Promise<IDefect | null> {
+    if (!body._id) throw new BadRequestError(`Missing Defect ID`);
+
+    const defect = await this.defectRepo.findById(body._id);
+
+    if (!defect) throw new NotFoundError(`Cannot find defect with ID ${body._id}`);
+
+    const updatedDefect = await this.defectRepo.update(body._id, {
+      status: body.status ? body.status : defect.status,
+      description: body.description ? body.description : defect.description,
+    } as IDefect);
+    return updatedDefect;
+  }
+
```

Comment:
I can update a defect with a user-defined status, I think you just need to update the [DefectModel](https://github.com/Mackbellemore/soen-390-team07/blob/1d48c873c4fef9c34d0a5fe9a77ed88c08be0408/server/app/models/DefectModel.ts#L14) to only take an enum of strings.

Reply:
Good catch!!